quickwit/quickwit-metastore/src/metastore/mod.rs

```diff
@@ -336,6 +338,15 @@ pub struct ListSplitsQuery {
 
     /// The create timestamp range to filter by.
     pub create_timestamp: FilterRange<i64>,
+
+    /// Is the split mature or immature at a given datetime.
+    pub maturity: Option<SplitMaturityFilter>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct SplitMaturityFilter {
```

Comment:
nitpick:
`Bound<OffsetDateTime>` instead of `Option<SplitMaturityFilter>` could have also done the job in an unambiguous manner.